### PR TITLE
Feature/DL-43

### PR DESCRIPTION
**Pull request by mssliw**
Deleting library items enabled.

- template for the removal confirmation created;
- only the admin has privileges to delete any item or copy;
- while multiple copies of an item exist - it is possible to remove every copy or just the selected one (works only for library items with asset code);
- after removing an item or a copy, short message is flashed to confirm that it has been removed;
- after removing  an item or a copy, there is no possibility to load the template of item removal (404).

### CODE REVIEW

File: templates/remove_item.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                                            {{ table_row('Title', item.title) }}
+                                            {% if item.type == 'magazine' %}
+                                                {{ table_row('Year', item.year) }}
+                                                {{ table_row('Issue', item.issue) }}
+                                            {% elif item.type == 'book' %}
+                                                {{ table_row('Authors', authors_list) }}
+                                            {% else %}
+                                                <p class="error-info"> There occurred some problems with loading library item. </p>
+                                            {% endif %}
+
```
[the remark this answers is not in the document]
Comment (mssliw) [reply templates/remove_item.html]: yep, indeed!

File: tests/test_remove_item.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    populate_books)
+
+
+def test_remove_item(session, db_user, client):
+    user = db_user
+    books = populate_books(n=5)
+    session.add_all(books)
+    session.commit()
+    copies_before = session.query(Copy).count()
+    for b in books:
```
[the remark this answers is not in the document]
Comment (mssliw) [reply tests/test_remove_item.py]: Thanks, I took care of it.

**Comment by kuszcjan** [comment]
I would merge this one just after https://github.com/InkubatorTieto/melvil/pull/53

File: views/index.py [hunk 3]
```diff
@@ -325,6 +327,59 @@ def reserve(copy_id):
...
+    try:
+        user = User.query.get(session['id'])
+        admin = user.has_role('ADMIN')
+    except KeyError:
+        abort(401)
+    except Exception:
+        abort(500)
+    form = RemoveForm()
+    item = LibraryItem.query.get_or_404(item_id)
+    copy = Copy.query.filter_by(asset_code=asset_code).first_or_404()
```
Comment (oskam) [inline views/index.py]: Not better just pass to the view `copy.id`? As there could be copies without assets, plus `query.get(id)` is more efficient approach than `query.filter_by(...)`

Comment (mssliw) [reply views/index.py]: Yep, I've tried to handle it this way, but had some obstacles. 
While the asset code does not exist (which tends to happen while we have only one copy of a book or magazine), the whole item is removed, so in the end I think it works properly.

File: views/index.py [hunk 4]
```diff
@@ -440,8 +440,8 @@ def item_description(item_id):
     return render_template('item_description.html',
                            item=item,
                            tags_list=tags_list,
-                           authors_list=authors_list,
-                           admin=admin)
+                           authors_list=authors_list)
+                           # admin=admin)
```
Comment (oskam) [inline views/index.py]: Same as above.

File: views/index.py [hunk 5]
```diff
@@ -423,13 +423,13 @@ def add_like(wish_id):
...
-        abort(401)
-    except Exception:
-        abort(500)
+    # try:
+    #     user = User.query.get(session['id'])
+    #     admin = user.has_role('ADMIN')
+    # except KeyError:
+    #     abort(401)
+    # except Exception:
+    #     abort(500)
```
Comment (oskam) [inline views/index.py]: We shouldn't pass comments like that into the release

File: views/index.py [hunk 6]
```diff
@@ -364,7 +364,7 @@ def remove_copy(item_id, asset_code):
         abort(500)
     form = RemoveForm()
     item = LibraryItem.query.get_or_404(item_id)
-    copy = Copy.query.filter_by(asset_code=asset_code).first_or_404()
+    copy = Copy.query.filter_by(id=copy_id).first_or_404()
```
Comment (oskam) [inline views/index.py]: Or just `Copy.query.get_or_404(copy_id)`

File: tests/test_remove_item.py [hunk 7]
```diff
@@ -6,24 +6,24 @@
...
-
         for copy in copies:
             session.delete(copy)
             session.commit()
-            resp = client.post(url_for('library.remove_item',
-                                       item_id=copy.id,
+            resp = client.post(url_for('library.remove_copy',
+                                       item_id=book.id,
+                                       copy_id=copy.id,
                                        db_user=user))
```
Comment (oskam) [inline tests/test_remove_item.py]: I think this line is unnecessary.

**Pull request merged**